### DISCUSSION TITLE
DragValue not support empty content in text editing mode

### DISCUSSION THREAD

**Discussion by tkkcc** [emilk/egui#2376] [General]
A normal way to edit `2` into `0` is `backspace` + `0`. Current DragValue does not support this.

**Comment by xTibor** [comment]
This is a regression introduced somewhere between `0.19.0` and `master`. There are also other issues editing `DragValue`s on `master`:

- Cannot enter decimal points after numbers anymore. For example you cannot turn "0" into "0.1" because `DragValue` swallows those decimal points.

- Entering negative signs in front of numbers are broken. You cannot turn "0" into "-0.1" because `DragValue` swallows negative signs in front of zeroes. It started eagerly and incorrectly normalizing negative zero into zero while in edit mode.

**Reply by xTibor** [reply]
PR #2342 has caused this regression.
